_posts/2019-10-27-on-code-reviews.md

```diff
@@ -0,0 +1,87 @@
+reviews. If there's something you like or learned, let the author know. I
+believe this will make a big difference on how a review is perceived, and is a
+win-win all around.<br />
+This is what I believe to be the **single most important part of a review**.
+- **I read the pull request's description _and_ ticket.** Yes, also the ticket.
+Seldom does that not help me getting a better understanding of the code. And
+understanding is key. This makes everyone involved's time a lot more valuable.
+If something in the description or ticket is unclear to me, I try to reach out
+the the author before continuing a review.
+- **I provide context.** If I come across something that "might better be
+represented by a function rather than a computed property", I explain why I
+would suggest that in the given case.
+- **I provide references.** Whether it's a blog post, piece of code, or
+documentation I reference, I link it. This allows the author to dig deeper into
+the topic if they want, and may be helpful for those who read the comment(s) in
+the future.
+- **I take the time to try and understand the code.** When I feel like I'm
+having a hard time grasping some of the decisions that have been made, or the
+code that has been written in this pull request, I often feel asking about it
+in writing is difficult. If this happens, I either take a step away from the
+computer and try again later, or ask to go over it in person (face-to-face or
+in a video call, _not_ a phone call. Being able to see each other aids
+understanding.)
+- **I ask questions.** I would call myself a curious person. If I see something
+I don't know about, or want to know more about (eg. a piece of code, business
+logic), I ask. If the review already has many comments, I might do this
+outside the review itself.
+- **I re-read your review before submitting it.** I don't always do this, but I
+know it helps. I've seen the author ask about comments I wrote, and having a
+hard time understanding what I meant. I feel that this has been happening less
+and less over time, though! But communication is hard.
+- **I feel responsibility for the code I review,** meaning I use "we", not
+"you". "We could consider adding a comment explaining why we need to ignore X",
+not "you".
+- **I review my own code.** Especially in larger pull requests, I look through
+the code before assigning a reviewer, either asking open questions to the
+reviewer, making notes and/or making some changes. Seeing the "bigger" picture
+often helps me catch issues/improvements to be made early, hopefully making the
+code review more enjoyable in the process.
+
+---
+
+Keep in mind that this is not an exhaustive list, but rather a bunch of the
+principles I try to keep in mind while reviewing. :)
+
+Bonus: I'm a big fan of a tool we built at work to get a random reviewer for
+your code. It's called `lotto`, and picks a random reviewer outside of your
+team. It's great.
+
```

Comment:
Do you have something you still haven’t tried or have seen that it works for others?

Basically showing that you are still learning, want people to reach out with ideas or suggestions.

Reply:
Good point; added!